profitbricks/src/main/java/org/jclouds/profitbricks/domain/LoadBalancer.java

```diff
@@ -0,0 +1,398 @@
+                 .ip(in.ip()).lanId(in.lanId()).state(in.state()).creationTime(in.creationTime()).lastModificationTime(in.lastModificationTime()).balancedServers(in.balancedServers()).firewalls(in.firewalls());
+      }
+   }
+
+   public static final class Request {
+
+      public static CreatePayload.Builder creatingBuilder() {
+         return new CreatePayload.Builder();
+      }
+
+      public static UpdatePayload.Builder updatingBuilder() {
+         return new UpdatePayload.Builder();
+      }
+
+      public static RegisterPayload.Builder registerBuilder() {
+         return new RegisterPayload.Builder();
+      }
+
+      public static DeregisterPayload.Builder deregisterBuilder() {
+         return new DeregisterPayload.Builder();
+      }
+
+      @AutoValue
+      public abstract static class CreatePayload {
+
+         public abstract String dataCenterId();
+
+         public abstract String loadBalancerName();
+
+         public abstract LoadbalancerAlgorithm loadBalancerAlgorithm();
+
+         public abstract String ip();
+
+         @Nullable
+         public abstract String lanId();
+
+         @Nullable
+         public abstract List<String> serverIds();
+
+         public static CreatePayload create(String dataCenterId, String loadBalancerName, LoadbalancerAlgorithm loadBalancerAlgorithm, String ip, String lanId, List<String> serverIds) {
+            return new AutoValue_LoadBalancer_Request_CreatePayload(dataCenterId, loadBalancerName, loadBalancerAlgorithm, ip, lanId, serverIds);
+         }
+
+         public static class Builder {
+
+            public String dataCenterId;
+            public String loadBalancerName;
+            public LoadbalancerAlgorithm loadBalancerAlgorithm;
+            public String ip;
+            public String lanId;
+            public List<String> serverIds;
+
+            public Builder dataCenterId(String dataCenterId) {
+               this.dataCenterId = dataCenterId;
+               return this;
+            }
+
+            public Builder loadBalancerName(String loadBalancerName) {
+               this.loadBalancerName = loadBalancerName;
+               return this;
+            }
+
+            public Builder loadBalancerAlgorithm(LoadbalancerAlgorithm loadBalancerAlgorithm) {
+               this.loadBalancerAlgorithm = loadBalancerAlgorithm;
+               return this;
+            }
+
+            public Builder ip(String ip) {
+               this.ip = ip;
+               return this;
+            }
+
+            public Builder lanId(String lanId) {
+               this.lanId = lanId;
+               return this;
+            }
+
+            public Builder serverIds(List<String> serverIds) {
+               this.serverIds = serverIds;
+               return this;
+            }
+
+            public CreatePayload build() {
+               return CreatePayload.create(dataCenterId, loadBalancerName, loadBalancerAlgorithm, ip, lanId, serverIds);
```

Comment:
Add the IP validation here as well. (So that we can preemptively fail users attempting to provision with invalid IP)